script/validators/source_validator.py

```diff
@@ -219,8 +219,35 @@ def check_namespaces(file_path):
     return status
 
 
+def check_includes(file_path):
+    """Checks whether local includes are done via #include<...>"""
+    with open(file_path, "r") as f:
+        path_pattern = re.compile(r'<(.*?)>')
+        includes = os.listdir('src/include')
+        linenum = 0
+        status = True
+        for line in f:
+            linenum += 1
+            if line.startswith('#include <'):
+                path = path_pattern.search(line).groups()[0]
+                path = path.split('/')
+                word = path[0]
+                if word == 'include':
+                    word = path[1]
+                if word in includes:
```

Comment:
Be aware that if you only check the first directory, we might get false positives. For now it works, but if you want to make it more fancy, create a set with *all* files of /src/include at the beginning of the script and check if the path exists in this list (path without `include/`).

Reply:
_All_ means all the possible paths of `src/include`?

Reply:
It's a lot, I know. (474, see below) But if we put them in a set (not a list) I hope that python does a fast lookup. And once at the beginning of the script, not for every file. I think you can use `glob` for that. 

```
$ find ./src/include/ -type f | wc -l
474
```

Reply:
I used `glob` and I'm getting 470. Perhaps because I traversed only the paths ending with `.h`?